src/components/hooks/usePlayerPreviewHeight.js

```diff
@@ -0,0 +1,23 @@
+import { useLayoutEffect, useState, useRef } from 'react'
+
```

Comment:
I have no idea what you are doing in this file, can you explain it to me?  🤷‍♂️

Reply:
Both this and the useEnemyPreviewHeight hook are helper functions to calculate the individual height of my preview components, so the show/hide animation works smoothly. Did that help? If not just ask me on Slack :)